raft/backend.go

```diff
@@ -36,7 +36,7 @@ type RaftService struct {
 	nodeKey  *ecdsa.PrivateKey
 }
 
-func New(ctx *node.ServiceContext, chainConfig *params.ChainConfig, raftId, raftPort uint16, joinExisting bool, blockTime time.Duration, e *eth.Ethereum, startPeers []*enode.Node, datadir string) (*RaftService, error) {
+func New(ctx *node.ServiceContext, chainConfig *params.ChainConfig, raftId, raftPort uint16, joinExisting bool, blockTime time.Duration, e *eth.Ethereum, startPeers []*enode.Node, datadir string, ethCfg *eth.Config) (*RaftService, error) {
```

Comment:
changing function signature is ok but in this case it seems the logic could be encapsulated

`eth.Ethereum` has the `eth.Config`. I think it'd be better to create a function in `eth.Ethereum` to do `CalcGasLimit` and inject into the `RaftService` so `raft/minter.go` can utilize that function when `createWork()`

my 2 cents